be/src/util/priority_thread_pool.hpp

```diff
@@ -123,7 +125,50 @@ class PriorityThreadPool {
         join();
     }
 
+    void set_num_thread(int num_thread) {
+        size_t num_thread_in_pool = _threads.size();
+        if (num_thread > num_thread_in_pool) {
+            increase_thr(num_thread - num_thread_in_pool);
+        } else if (num_thread < num_thread_in_pool) {
+            decrease_thr(num_thread_in_pool - num_thread);
+        }
+    }
+
 private:
+    void increase_thr(int num_thread) {
+        std::lock_guard<std::mutex> l(_lock);
+        for (int i = 0; i < num_thread; ++i) {
+            new_thread(++_current_thread_id);
+        }
+    }
+
+    void decrease_thr(int num_thread) {
+        std::lock_guard<std::mutex> l(_lock);
+        for (int i = 0; i < num_thread; ++i) {
+            _should_decrease++;
+        }
+        for (int i = 0; i < num_thread; ++i) {
+            PriorityThreadPool::Task empty_task = {0, []() {}};
+            _work_queue.try_put(empty_task);
+        }
+    }
+
+    // not thread safe
+    // we need acquire _lock before call this function
+    void new_thread(int tid) {
+        auto* thr = _threads.create_thread(std::bind<void>(std::mem_fn(&PriorityThreadPool::work_thread), this, tid));
+        _threads_holder.emplace_back(thr, tid);
+    }
+
+    void remove_thread(int tid) {
+        auto res = std::find_if(_threads_holder.begin(), _threads_holder.end(),
+                                [=](const auto& val) { return tid == val.second; });
+        if (res != _threads_holder.end()) {
+            _threads.remove_thread(res->first);
+            _threads_holder.erase(res);
```

Comment:
std::vector is not thread-safety, maybe erase will be called concurrenctly?

Reply:
Sorry I forget add a guard here